cloudant-sync-datastore-core/src/main/java/com/cloudant/sync/replication/BasicPullStrategy.java

```diff
@@ -179,6 +180,36 @@ public void run() {
         
     }
 
+    // Inner class used to fetch and process a batch of changes. See comment in replicate() for
+    // more details.
+    private class BatchProcessor implements Callable<BatchProcessor> {
```

Comment:
The returned `BatchProcessor` doesn't appear to be used (i.e. there is no assignment after the `call()`, is there any reason not to make this a `Callable<Void>`? Is there some future use case in mind?